### PR TITLE
Update `lists::contains` to experimental row comparator

### PR DESCRIPTION
**Pull request by divyegala**
## Description
Part of https://github.com/rapidsai/cudf/issues/11844

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by davidwendt** [comment]
Looks like `src/lists/contains.cu` now takes over an hour to compile on it's own with this change.
71 minutes: https://downloads.rapids.ai/ci/cudf/pull-request/13672/e6059be/cuda12_x86_64.ninja_log.html 
86 minutes: https://downloads.rapids.ai/ci/cudf/pull-request/13672/e6059be/cuda11_aarch64.ninja_log.html

**Comment by mythrocks** [comment]
This change looks good to me, barring the concern regarding the increased compile time. :/

**Review by mythrocks — APPROVED** [review]

File: cpp/src/lists/contains.cu [hunk 1]
```diff
@@ -104,62 +104,21 @@ __device__ auto element_index_pair_iter(size_type const size)
...
-  }
-};
-
 /**
  * @brief Functor to perform searching for index of a key element in a given list, specialized
  * for nested types.
  */
 template <typename KeyValidityIter, typename EqComparator>
-struct search_list_nested_types_fn {
+struct search_list {
```
Comment (ttnghia) [inline cpp/src/lists/contains.cu:112]: There is also the unused code remaining at the top of this file.

**Review by ttnghia — CHANGES_REQUESTED** [review]
Please provide benchmarks for supporting the changes.

**Pull request closed without merging**